src/librustc/session/config.rs

```diff
@@ -1715,6 +1704,22 @@ pub fn build_session_options_and_crate_config(
         ),
     };
 
+    let edition = match matches.opt_str("edition") {
+        Some(arg) => match Edition::from_str(&arg){
+            Ok(edition) => edition,
+            Err(_) => early_error(
+                ErrorOutputType::default(),
+                &format!(
+                    "argument for --edition must be one of: \
```

Comment:
We also need to handle stability here. If the picked edition is unstable, we should throw an error if the compiler is not nightly.

You can add a `.stable()` method to `Edition` that returns true for 2015.

Reply:
will do

Reply:
How do I know if the compiler is nightly?

Reply:
nvm, found it